relm4/src/typed_list_view/mod.rs

```diff
@@ -0,0 +1,424 @@
+            let value = get_value::<T>(obj);
+            f(&value)
+        });
+        let filter_model =
+            gtk::FilterListModel::new(Some(self.active_model.clone()), Some(filter.clone()));
+        self.active_model = filter_model.clone().upcast();
+        self.selection_model.set_list_model(&self.active_model);
+        self.filters.push(Filter {
+            filter,
+            model: filter_model,
+        });
+    }
+
+    /// Returns the amount of filters that were added.
+    pub fn filters_len(&self) -> usize {
+        self.filters.len()
+    }
+
+    /// Set a certain filter as active or inactive.
+    pub fn set_filter_status(&mut self, idx: usize, active: bool) -> bool {
+        if let Some(filter) = self.filters.get(idx) {
+            if active {
+                filter.model.set_filter(Some(&filter.filter));
+            } else {
+                filter.model.set_filter(None::<&gtk::CustomFilter>);
+            }
+            true
+        } else {
+            false
+        }
+    }
+
+    /// Remove the last filter.
+    pub fn pop_filter(&mut self) {
+        let filter = self.filters.pop();
+        if let Some(filter) = filter {
+            self.active_model = filter.model.model().unwrap();
+            self.selection_model.set_list_model(&self.active_model);
+        }
+    }
+
+    /// Remove all filters.
+    pub fn clear_filters(&mut self) {
+        self.filters.clear();
+        self.active_model = self.base_model.clone();
+        self.selection_model.set_list_model(&self.active_model);
+    }
+
+    /// Add a new item at the end of the list.
+    pub fn append(&mut self, value: T) {
+        self.store.append(&glib::BoxedAnyObject::new(value));
+    }
+
+    /// Add new items from an iterator the the end of the list.
+    pub fn extend_from_iter<I: IntoIterator<Item = T>>(&mut self, init: I) {
+        let objects: Vec<glib::BoxedAnyObject> =
+            init.into_iter().map(glib::BoxedAnyObject::new).collect();
+        self.store.extend_from_slice(&objects);
+    }
+
+    #[cfg(feature = "gnome_43")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "gnome_43")))]
+    /// Find the index of the first item that matches a certain function.
+    pub fn find<F: FnMut(&T) -> bool>(&self, mut equal_func: F) -> Option<u32> {
+        self.store.find_with_equal_func(move |obj| {
+            let value = get_value::<T>(obj);
+            equal_func(&value)
+        })
+    }
+
+    /// Returns true if the list is empty.
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    /// Returns the length of the list (without filters).
+    pub fn len(&self) -> u32 {
+        self.store.n_items()
+    }
+
+    /// Get the [`TypedListItem`] at the specified position.
+    ///
+    /// Returns [`None`] if the position is invalid.
+    pub fn get(&self, position: u32) -> Option<TypedListItem<T>> {
+        if let Some(obj) = self.store.item(position) {
```

Comment:
This gets the position of the item within the initial input. However, when working with the Selection, it returns the position after sorting and filtering. If we use the active_model for the position, then this aligns with the ordering of the widgets as shown including sorting and filtering.

This might warrant it's own function within the API.

```suggestion
        if let Some(obj) = self.active_model.item(position) {
```